aten/src/ATen/native/mkldnn/xpu/detail/Attention.cpp

```diff
@@ -38,46 +44,45 @@ struct SDPALogicalParams {
   std::optional<logical_tensor> neg_inf;
   std::optional<logical_tensor> attn_mask;
   logical_tensor value{};
-  logical_tensor output{};
+  logical_tensor attention{};
+  std::optional<logical_tensor> logsumexp;
 
   SDPALogicalParams(
       const at::Tensor& query_,
       const at::Tensor& key_,
       const at::Tensor& value_,
       const std::optional<at::Tensor>& attn_mask_,
-      const at::Tensor& output_,
+      const at::Tensor& attention_,
+      const at::Tensor& logsumexp_,
       int batch_size,
       int seq_len_q,
       int seq_len_kv,
       int num_head_q,
       int num_head_kv,
       int head_dim_qk,
       int head_dim_v,
-      bool is_causal) {
+      bool is_causal,
+      bool compute_logsumexp) {
     const data_type dtype = to_logical_tensor_data_type(query_.scalar_type());
     TORCH_INTERNAL_ASSERT(
         (dtype != data_type::undef),
         "Only FP16/BF16/FP32 datatypes are currently supported");
+    TORCH_INTERNAL_ASSERT(
+        query_.scalar_type() == attention_.scalar_type(),
+        "scaled_dot_product_attention_xpu: query and attention tensors should have the same data type.");
+    TORCH_INTERNAL_ASSERT(
+        !at::native::onednn::is_broadcast(query_) &&
+            !at::native::onednn::is_broadcast(key_) &&
+            !at::native::onednn::is_broadcast(value_),
+        "scaled_dot_product_attention_xpu: tensors q/k/v should not be broadcasted tensor.");
     const dims scalar_shape = {1};
-    std::vector<logical_tensor> inputLogicalTensors;
 
     at::Tensor reshaped_query = query_;
     at::Tensor reshaped_key = key_;
     at::Tensor reshaped_value = value_;
-    at::Tensor reshaped_output = output_;
+    at::Tensor reshaped_attention = attention_;
+    at::Tensor reshaped_logsumexp = logsumexp_.unsqueeze(-1);
     at::Tensor reshaped_attn_mask = attn_mask_.value_or(at::Tensor());
-    if (at::native::onednn::is_broadcast(reshaped_query)) {
```

Comment:
With this code change, sdpa will not support broadcast anymore. Is this a BC breaking, any impact for the old script.